internal/operators/cnv/cnv_operator.go

```diff
@@ -0,0 +1,139 @@
+package cnv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/hardware"
+	"github.com/openshift/assisted-service/internal/operators/api"
+	models "github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+type operator struct {
+	log logrus.FieldLogger
+}
+
+// NewCNVOperator creates new instance of a Container Native Virtualization installation plugin
+func NewCNVOperator(log logrus.FieldLogger) *operator {
+	return &operator{
+		log: log,
+	}
+}
+
+// GetType reports type of an operator this Operator manages
+func (o *operator) GetType() models.OperatorType {
+	return models.OperatorTypeCnv
+}
+
+// GetDependencies provides a list of dependencies of the Operator
+func (o *operator) GetDependencies() []models.OperatorType {
+	return []models.OperatorType{models.OperatorTypeLso}
+}
+
+// GetClusterValidationID returns cluster validation ID for the Operator
+func (o *operator) GetClusterValidationID() string {
+	return string(models.ClusterValidationIDCnvRequirementsSatisfied)
+}
+
+// GetHostValidationID returns host validation ID for the Operator
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDCnvRequirementsSatisfied)
+}
+
+// ValidateCluster always return "valid" result
+func (o *operator) ValidateCluster(ctx context.Context, cluster *common.Cluster) (api.ValidationResult, error) {
+	var insufficientHosts []string
+	var status string
+	for _, host := range cluster.Hosts {
+		result, err := o.ValidateHost(ctx, cluster, host)
+		if err != nil {
+			return api.ValidationResult{Status: api.Failure, ValidationId: o.GetClusterValidationID()}, err
+		}
+		if result.Status != api.Success {
+			insufficientHosts = append(insufficientHosts, fmt.Sprint("Insufficient resources on host with host ID ", *host.ID, " to deploy CNV"))
+		}
+	}
+	if len(insufficientHosts) > 0 {
+		for _, hostStatus := range insufficientHosts {
+			status = status + hostStatus + ".\n"
+		}
+		err := setOperatorStatus(&cluster.Cluster, status)
```

Comment:
No need to set the operator status - it was a temporary solution for OCS until we will have the generic interface

Reply:
OK, will remove